web_ui/ui_test.go

```diff
@@ -496,3 +503,261 @@ func TestServerHostRestart(t *testing.T) {
 		}
 	})
 }
+
+func TestApiToken(t *testing.T) {
+	route := gin.New()
+	route.POST("/api/v1.0/createToken", createApiToken)
+	route.DELETE("/api/v1.0/deleteApiToken/:id", deleteApiToken)
+	route.GET("/privilegedRoute", func(ctx *gin.Context) {
+		authOption := token.AuthOption{
+			Sources: []token.TokenSource{token.Header},
+			Issuers: []token.TokenIssuer{token.APITokenIssuer},
+			Scopes:  []token_scopes.TokenScope{token_scopes.Monitoring_Scrape},
+		}
+
+		status, ok, err := token.Verify(ctx, authOption)
+		if !ok {
+			ctx.JSON(status, server_structs.SimpleApiResp{
+				Status: server_structs.RespFailed,
+				Msg:    err.Error(),
+			})
+			return
+		}
+	})
+
+	ctx, cancel, egrp := test_utils.TestContext(context.Background(), t)
+	defer func() { require.NoError(t, egrp.Wait()) }()
+	defer cancel()
+
+	tempFile, err := os.CreateTemp("", "web-ui-passwd-api-token")
+	if err != nil {
+		fmt.Println("Failed to setup web-ui-passwd file")
+		os.Exit(1)
+	}
+	defer os.Remove(tempFile.Name())
+
+	dirName := t.TempDir()
+	server_utils.ResetTestState()
```

Comment:
Best practice is to put this at the top of your test:
```
server_utils.ResetTestState()
defer server_utils.ResetTestState()
```

This makes sure you start clean and end clean.